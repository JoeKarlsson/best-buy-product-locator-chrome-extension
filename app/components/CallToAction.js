import React from 'react';

const CallToAction = () => (
  <div>
    Want it faster? This same product is available at Best Buy. Order now and pick it up in 1 hour.
    Price: $129.99
  </div>
);

export default CallToAction;
