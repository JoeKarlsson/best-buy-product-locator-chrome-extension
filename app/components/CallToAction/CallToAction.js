import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { COLORS } from '../../constants/styles';

const Container = styled.div`
  ${(props) => {
    if (!props.isPopup) {
      return css`
        margin-left: 100px;
        margin-top: -5px;

        p {
          display: inline;
        }

        @media (max-width: 1200px) {
          margin-top: 5px;
          margin-left: 0px;
        }

        @media (max-width: 850px) {
          p {
            display: block;
          }
        }
      `;
    }
  }};
`;

const H2 = styled.h2`
  color: ${COLORS.darkBlue};
`;

const CallToAction = (props) => {
  const {
    isPopup, nearestStoreMapUrl, nearestStore, price
  } = props;
  return (
    <Container isPopup={isPopup}>
      <H2 className="foobar">Want it faster? </H2>
      <p>
        This product is available for pick up at the{' '}
        <a target="_blank" rel="noopener noreferrer" href={nearestStoreMapUrl}>
          Best Buy in {nearestStore}.
        </a>{' '}
        Order now and pick it up in 1 hour.&nbsp;
      </p>
      <p>
        Price: <b>${price}</b>
      </p>
    </Container>
  );
};

CallToAction.propTypes = {
  isPopup: PropTypes.bool,
  nearestStore: PropTypes.string.isRequired,
  nearestStoreMapUrl: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
};

CallToAction.defaultProps = {
  isPopup: false,
};

export default CallToAction;
