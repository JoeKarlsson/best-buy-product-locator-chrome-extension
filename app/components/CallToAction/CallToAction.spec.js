import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import CallToAction from './CallToAction';

configure({ adapter: new Adapter() });

describe('CallToAction', () => {
  let wrapper;

  const hours = {
    closingSoon: true,
    openNow: true,
    today: {
      close: '21:00',
      closeAmPm: '9:00 pm',
      date: '2018-08-27',
      day: 'Today',
      open: '10:00',
      openAmPm: '10:00 am',
    },
    tomorrow: {
      close: '21:00',
      closeAmPm: '9:00 pm',
      date: '2018-08-28',
      day: 'Tomorrow',
      open: '10:00',
      openAmPm: '10:00 am',
    },
  };

  beforeEach(() => {
    wrapper = shallow(
      <CallToAction
        hours={hours}
        nearestStore="Richfeild, MN"
        nearestStoreMapUrl="www.testurl.com"
        price={1234}
      />,
    );
  });

  describe('rendering', () => {
    describe('initial state', () => {
      it('is rendered', () => {
        const component = renderer.create(
          <CallToAction
            hours={hours}
            nearestStore="Richfeild, MN"
            nearestStoreMapUrl="www.testurl.com"
            price={1234}
          />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('is rendered correctly', () => {
        expect(wrapper).toHaveLength(1);
      });
    });
  });
});
