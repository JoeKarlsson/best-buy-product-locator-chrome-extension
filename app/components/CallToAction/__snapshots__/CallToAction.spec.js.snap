// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction rendering initial state is rendered 1`] = `
<div
  className="sc-bdVaJa kZuZoP"
>
  <h2
    className="foobar sc-bwzfXH gPMGtq"
  >
    Want it faster? 
  </h2>
  <p>
    This product is available for pick up at the
     
    <a
      href="www.testurl.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Best Buy in 
      Richfeild, MN
      .
    </a>
     
    Order now and pick it up in 1 hour.Â 
  </p>
  <p>
    Price: 
    <b>
      $
      1234
    </b>
  </p>
</div>
`;
